### ISSUE TITLE
Alter source schema by refreshing schema registry

### ISSUE THREAD

**Issue by Rossil2012** [risingwavelabs/risingwave#14056]
Track https://github.com/risingwavelabs/risingwave/pull/14057.

**Comment by xiangjinwu** [comment]
Has this been supported? Could you link the PR?

**Comment by Rossil2012** [comment]
This issue is created by a misclick in #11800.
The work is on going (see #14057).
Let's reopen it as a tracking issue anyway.

**Comment by st1page** [comment]
> I wonder what is the ultimate syntax for this feature? The protobuf message name can be unchanged when updating the schema in the most common case.
> 
> @st1page

`ALTER SOURCE xx FORMAT xx ENCODE xx (...)`
e.g. 
```
   ALTER SOURCE xx FORMAT xx ENCODE xx (...) FORMAT plain ENCODE protobuf(
        schema.registry = 'http://message_queue:8081',
        message = 'test.User'
    );
```
The encode options will overwrite the old one and re-registry the schema 
Altering data_format and data_encode is not supported, in other words, they should be the same the current.

**Comment by Rossil2012** [comment]
Closed as #14057 is merged. `alter table with conn` is tracked in #11800.
